app/src/main/java/com/duckduckgo/app/browser/httpauth/WebViewHttpAuthStore.kt

```diff
@@ -75,7 +76,7 @@ interface WebViewHttpAuthStore {
 )
 @SingleInstanceIn(AppScope::class)
 class RealWebViewHttpAuthStore @Inject constructor(
-    private val webViewDatabase: WebViewDatabase,
+    private val webViewDatabase: Provider<WebViewDatabase>,
```

Comment:
If we want to ensure that everybody gets a provider of the database, what do you think about modifying the signature of the dagger provider method from `fun provideWebviewDatabase(context: Context): WebViewDatabase` to `fun provideWebviewDatabase(context: Context): Provider<WebViewDatabase>` ??

This way nobody can request for the concrete `WebViewDatabase` type.